gcc/config/arc/arc.md

```diff
@@ -709,11 +712,126 @@
    (set_attr "predicable" "yes,no,yes,no,yes,no,yes,yes,yes,no,no,no,no,no,no,no")
    (set_attr "cpu_facility" "*,*,arcv1,em,*,*,*,*,*,*,*,*,*,*,em,*")])
 
+(define_insn "add2_jlioff"
+  [
+    (unspec_volatile [
+      (match_operand:SI 0 "move_dest_operand" "")
+      (match_operand:SI 1 "symbolic_operand" "")
+    ] VUNSPEC_ARC_ADD2_JLIOFF)
+  ]
+  ""
+  "add2_jlioff %0, %0, %1"
+  [(set_attr "type" "unary")
+  (set_attr "iscompact" "true")
+  (set_attr "predicable" "yes")
```

Comment:
Thank you. I'll fix that. Wrapping my head around the GCC machine description has been a challenge.